src/coreclr/nativeaot/Runtime/eventpipe/ep-rt-aot.h

```diff
@@ -1030,9 +1073,41 @@ ep_rt_temp_path_get (
     uint32_t buffer_len)
 {
     STATIC_CONTRACT_NOTHROW;
-//    EP_UNREACHABLE ("Can not reach here");
 
+#ifdef TARGET_UNIX
+
+	EP_ASSERT (buffer != NULL);
+	EP_ASSERT (buffer_len > 0);
+
+    const ep_char8_t *path = getenv ("TMPDIR");
+    if (path == NULL){
+        path = getenv ("TMP");
+        if (path == NULL){
+            path = getenv ("TEMP");
+            if (path == NULL)
+                path = "/tmp/";
+        }
+    }
+
+	int32_t result = snprintf (buffer, buffer_len, "%s", path);
+	if (result <= 0 || (uint32_t)result > buffer_len)
+		ep_raise_error ();
+
+	if (buffer [result - 1] != '/') {
+		buffer [result++] = '/';
+		buffer [result] = '\0';
```

Comment:
This has the potential of writing past the end of the buffer (if the buffer was exactly long enough to hold the value).